packages/taiko-client/prover/proof_producer/sgx_producer.go

```diff
@@ -117,13 +128,157 @@ func (s *SGXProofProducer) RequestProof(
+	log.Info(
+		"Aggregate sgx batch proofs from raiko-host service",
+		"items", items,
+	)
+	if len(items) == 0 {
+		return nil, errInvalidLength
+	}
+
+	blockIDs := make([]*big.Int, len(items))
+	for i, item := range items {
+		blockIDs[i] = item.Meta.GetBlockID()
+	}
+	batchProof, err := s.requestBatchProof(
+		ctx,
+		blockIDs,
+		items[0].Opts.ProverAddress,
+		items[0].Opts.Graffiti,
+		requestAt,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	metrics.ProverSgxProofAggregationGeneratedCounter.Add(1)
+
+	return &BatchProofs{
+		Proofs:     items,
+		BatchProof: batchProof,
+		Tier:       s.Tier(),
+	}, nil
+}
+
+// RequestCancel implements the ProofProducer interface to cancel the proof generating progress.
 func (s *SGXProofProducer) RequestCancel(
 	_ context.Context,
 	_ *ProofRequestOptions,
 ) error {
 	return nil
 }
 
+// requestBatchProof poll the proof aggregation service to get the aggregated proof.
+func (s *SGXProofProducer) requestBatchProof(
+	ctx context.Context,
+	blockIDs []*big.Int,
+	proverAddress common.Address,
+	graffiti string,
+	requestAt time.Time,
+) ([]byte, error) {
+	var (
+		proof []byte
+	)
+
+	ctx, cancel := rpc.CtxWithTimeoutOrDefault(ctx, s.RaikoRequestTimeout)
+	defer cancel()
+
+	blocks := make([][2]*big.Int, len(blockIDs))
+	for i, blockID := range blockIDs {
+		blocks[i][0] = blockID
+	}
+	reqBody := RaikoRequestProofBodyV3{
+		Type:     s.ProofType,
+		Blocks:   blocks,
+		Prover:   proverAddress.Hex()[2:],
+		Graffiti: graffiti,
+		SGX: &SGXRequestProofBodyParam{
+			Setup:     false,
+			Bootstrap: false,
+			Prove:     true,
+		},
+	}
+
+	client := &http.Client{}
+
+	jsonValue, err := json.Marshal(reqBody)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debug(
+		"Send batch proof generation request",
+		"blockIDs", blockIDs,
+		"proofType", "sgx",
+		"output", string(jsonValue),
+	)
+
+	req, err := http.NewRequestWithContext(
+		ctx,
+		"POST",
+		s.RaikoHostEndpoint+"/v3/proof/aggregate",
+		bytes.NewBuffer(jsonValue),
+	)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	if len(s.JWT) > 0 {
+		req.Header.Set("Authorization", "Bearer "+base64.StdEncoding.EncodeToString([]byte(s.JWT)))
+	}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Body.Close()
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to request batch proof, ids: %v, statusCode: %d", blockIDs, res.StatusCode)
+	}
+
+	resBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debug(
+		"Batch proof generation output",
+		"blockIDs", blockIDs,
+		"proofType", "sgx",
+		"output", string(resBytes),
+	)
+
+	var output RaikoRequestProofBodyResponseV2
+	if err := json.Unmarshal(resBytes, &output); err != nil {
+		return nil, err
+	}
+
+	if len(output.ErrorMessage) > 0 {
+		return nil, fmt.Errorf("failed to get batch proof, msg: %s", output.ErrorMessage)
+	}
+
+	if len(output.Data.Proof.Proof) == 0 {
```

Comment:
Make sure `output.Data` or `output.Data.Proof` is not null?

Reply:
I think doing this is enough.